mautic/mautic
# Fix segment mails not using site_url

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #10137 and #10042 



#### Description:

Based on https://github.com/mautic/mautic/pull/11492, plus test



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Change the site URL config to something else
3. Create a Segment
4. Create a Segment Email
5. Send the Segment Email
6. Check the email unsubscribe link. It should use the site_url instead of the current url you are on
